src/pybind/mgr/pg_autoscaler/module.py

```diff
@@ -321,9 +321,11 @@ def set_noautoscale(self) -> Tuple[int, str, str]:
     def serve(self) -> None:
         self.config_notify()
         while not self._shutdown.is_set():
-            if not self.has_noautoscale_flag():
-               self._maybe_adjust()
-               self._update_progress_events()
+            if not self.noautoscale:
```

Comment:
@kamoltat please check that again, it looks like in main we do have noautoscale and in quincy we are using the has_noautoscale_flag

Reply:
Great catch @NitzanMordhai I must have been asleep while resolving the merge conflict of this backport.